cmd/juju/application/utils/utils.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package utils
+
+import (
+	"github.com/juju/charm/v7"
+	charmresource "github.com/juju/charm/v7/resource"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/resource"
+)
+
+var logger = loggo.GetLogger("juju.cmd.juju.application.utils")
+
+// GetMetaResources retrieves metadata resources for the given
+// charm.URL.
+func GetMetaResources(charmURL *charm.URL, client CharmClient) (map[string]charmresource.Meta, error) {
+	charmInfo, err := client.CharmInfo(charmURL.String())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return charmInfo.Meta.Resources, nil
+}
+
+// ParsePlacement validates provided placement of a unit and
+// returns instance.Placement.
+func ParsePlacement(spec string) (*instance.Placement, error) {
+	if spec == "" {
+		return nil, nil
+	}
+	placement, err := instance.ParsePlacement(spec)
+	if err == instance.ErrPlacementScopeMissing {
+		spec = "model-uuid" + ":" + spec
```

Comment:
I know you didn't write this, but this makes my eyes bleed:


```suggestion
		spec = fmt.Sprintf("model-uuid:%s", spec)
```